### ISSUE TITLE
user-id in url

### ISSUE THREAD

**Issue by krzysiekbielicki** [allegro/restapi-guideline#4]
https://github.com/allegro/restapi-guideline#urls-for-users-collections notes that user collections must include user-id in URI.
What is the reason of this requirement?
URI should be designed upon responsibility of the service.
MyAccount service is responsible for retrieving authorized user data (identified by some Authorization info) and therefore userId in URI is redundant.
There may be some applications (for example OffersService) which may retrieve any users offers upon authorized or unauthorized request.

Is caching a concern here?
Neither /userinfo/{user-id} (with header) nor /userinfo (with header) can be cached becouse token (oauth) user:pass pair (basic auth) can be revoked and cache/proxy won't have any info abut that fact.

**Comment by walec51** [comment]
An URI should identify any resource uniquely - you want an URI that doesn't identify anything, your header would actually determine which userinfo resource is returned - this is not RESTfull and even violates the HTTP RFC.

As for your second argument the fact that authorization is required to access a specific userinfo doesn't mean it cannot be cached with a short TTL like 30 seconds for example. Many systems cache user roles and permissions with a short TTL for performance. Yes, changing user roles/permission has an 30 second propagation latency then but usually acceptable.

By the way what if you want an admin to have access to the userinfo resource of any user?

In summary omitting the user id in this case is considered a bad practice from a HTTP / REST stand point because it renders many caching and access features unimplementable.
